### PR TITLE
Feature: Change/Reset password API

### LINKED ISSUE #101: Change/Reset password
Almost forgot!

### PR DESCRIPTION
**Pull request by AlirezaT99**
closes #101 
- TODO
  - [x] Add documentation for APIs
  - [ ] Become synced with the frontend team about the emailed url in models.py

### CODE REVIEW

File: backend/api/models.py [hunk 1]
```diff
@@ -1,3 +1,24 @@
...
 # Create your models here.
+from django.dispatch import receiver
+from django_rest_passwordreset.signals import reset_password_token_created
+from django.core.mail import send_mail
+
+from templates.consts import *  
+
+
+@receiver(reset_password_token_created)
+def password_reset_token_created(sender, instance, reset_password_token, *args, **kwargs):
```
Comment (ahmadsalimi) [inline backend/api/models.py:13]: Why this code is in the `models` file? Is this model?

File: backend/api/models.py [hunk 2]
```diff
@@ -1,3 +1,24 @@
...
+
+    base_url = 'https://sharif-wss.ir/'  # TODO
+    email_message = "{}?token={}".format(base_url, reset_password_token.key)
```
Comment (ahmadsalimi) [inline backend/api/models.py:16]: I think the `base_url` should be passed to the `reset_password` API as a callback URL, like the payment procedure. It's not good for this to be hardcoded.

File: backend/templates/consts.py [hunk 3]
```diff
@@ -1,5 +1,7 @@
 PAYMENT_SUBJECT = 'تاییدیه پرداخت رویداد WSS'
-PAYMENT_HTML_CONTENT  = '''<!DOCTYPE html>
+RESET_PASSWORD_SUBJECT = 'WSS تنظیم مجدد رمز'
```
Comment (ahmadsalimi) [inline backend/templates/consts.py:2]: Shouldn't these subjects be in English? @emranbm

File: backend/templates/consts.py [hunk 4]
```diff
@@ -123,4 +121,16 @@
...
+    شرکت کننده‌ی محترم، {}<br />
+    با تشکر از پرداخت شما<br />
+    شماره‌ی پیگیری: {}
+'''
+
+RESET_PASSWORD_EMAIL = '''
+با سلام، برای تنظیم مجدد رمز خود
+<a href="{}">اینجا</a>
+کلیک کنید
+'''
```
Comment (ahmadsalimi) [inline backend/templates/consts.py:136]: And these email contents?

File: backend/api/models.py (hunk 1 above)
Comment (AlirezaT99) [reply backend/api/models.py:13]: It's not. The guy who wrote it had placed it in `models` =) works but is not appropriate. Apparently it's best practice to put _signal handlers_ in `api/signals/handlers.py`

File: backend/api/models.py (hunk 2 above)
Comment (AlirezaT99) [reply backend/api/models.py:16]: I agree. I'll see if I can override its view

File: backend/templates/consts.py (hunk 3 above)
Comment (emranbm) [reply backend/templates/consts.py:2]: Agree. and I think it applies to other texts like email body, etc.

File: backend/api/models.py (hunk 2 above)
Comment (AlirezaT99) [reply backend/api/models.py]: Will be discussed with the front team later

File: backend/api/models.py (hunk 1 above)
Comment (AlirezaT99) [reply backend/api/models.py]: placing it somewhere other than models or views fails to receive the signal and the approach i mentioned above failed in the first try.

File: backend/templates/consts.py (hunk 3 above)
Comment (AlirezaT99) [reply backend/templates/consts.py]: done

File: backend/templates/consts.py (hunk 4 above)
Comment (AlirezaT99) [reply backend/templates/consts.py]: done

**Review by emranbm — APPROVED** [review]

**Pull request merged**